tds_test.go

```diff
@@ -70,6 +70,206 @@ func TestSendLogin(t *testing.T) {
 	}
 }
 
+func TestParseLogin(t *testing.T) {
+	buf := NewTdsBuffer(1024, new(MockTransport))
+	originalLogin := login{
+		TDSVersion:     verTDS73,
+		PacketSize:     0x1000,
+		ClientProgVer:  0x01060100,
+		ClientPID:      100,
+		ClientTimeZone: -4 * 60,
+		ClientID:       [6]byte{0x12, 0x34, 0x56, 0x78, 0x90, 0xab},
+		OptionFlags1:   0xe0,
+		OptionFlags3:   8,
+		HostName:       "subdev1",
+		UserName:       "test",
+		Password:       "testpwd",
+		AppName:        "appname",
+		ServerName:     "servername",
+		CtlIntName:     "library",
+		Language:       "en",
+		Database:       "database",
+		ClientLCID:     0x204,
+		AtchDBFile:     "filepath",
+	}
+	err := sendLogin(buf, originalLogin)
+	if err != nil {
+		t.Error("sendLogin should succeed")
+	}
+
+	login, err := ReadLogin(buf)
+	if err != nil {
+		t.Error("ParseLogin should succeed")
+	}
+
+	// reset password, because this gets mangled before it's sent over the wire
+	originalLogin.Password = ""
+	login.login.Password = ""
+	//originalLogin.Password = string(manglePassword(originalLogin.Password))
+
+	if login.login.TDSVersion != originalLogin.TDSVersion {
+		fmt.Println("TDSVersion")
+		fmt.Println("Expected:", originalLogin.TDSVersion)
+		fmt.Println("Returned:", login.login.TDSVersion)
+		t.Error("input output don't match for TDSVersion")
```

Comment:
Assuming we can't figure out how to make deepEqual work (I think we can) we could at least package up these 4 repeated test assertions into a local func.

Reply:
I’m thinking to stick with the lengthy 1 by 1 assertions because when there’s a failure we have insight into what’s wrong, less so with DeepEqual.

Reply:
Curious... Did you have a chance to figure out why DeepEqual wasn't working in this case, or is this a `don't care` now?

Reply:
@diverdane It is because one of the fields is a byte slice. Explanation straight from the Go docs https://golang.org/pkg/reflect/#DeepEqual
> Note that a non-nil empty slice and a nil slice (for example, []byte{} and []byte(nil)) are not deeply equal.

Reply:
@doodlesbykumbi That repetition is too much.  How about you just use:

https://godoc.org/github.com/stretchr/testify/assert#EqualValues

This uses `DeepEqual` underneath, but also permforms and diff and reports that when the match fails.

Reply:
That'd require introducing a new dep to a fork. For only addressing this particular test case I don't think we'd be justified. This repo has minimal external deps and I think for now it's reasonable to stick with that

Reply:
@jonahx That'd require introducing a new dep to a fork. For only addressing this particular test case I don't think we'd be justified. This repo has minimal external deps and I think for now it's reasonable to stick with that

Reply:
Yeah, I can see your point there.  How about just a helper func then to reduce the boilerplate:

```go
func Fail(t testing.T, name string, expected, actual interface{}) {
	fmt.Println("Expected:", expected)
	fmt.Println("Returned:", actual)
	t.Errorf("input output don't match for %s", name)
}
```